ssw/alignmentmgr.pyx

```diff
@@ -78,17 +103,55 @@ cdef extern from "ssw.h":
 
 import warnings as pywarnings
 from typing import (
-    List,
     Optional,
     Tuple,
     Union,
 )
 
-from .alignment import Alignment
+from .alignmenttuple import Alignment
 
 SNAKE_CASE_DEPRECATED_MSG = 'Function deprecated please use "%s" instead'
 
 
+class BitwiseAlignmentFlag:
+    '''``bitwise_flag`` (from high- to low-bit) for :meth:`AlignmentMgr.align`
+
+    Attributes:
+        best_idxs_no_cigar: bit 5 when set as 1, ``ssw_align`` will return
+            the best alignment beginning position;
+            NOTE: this is setting ``bitwise_flag == 8``
+        distance_filter: bit 6 when set as 1::
+
+                if (
+                    ((reference_end - reference_start) < distance_filter) and
+                    ((read_end - read_start) < distance_filter)
+                )
+
+            (whatever bit 5 is set) the function will return the best
+            alignment beginning position and cigar;
+            NOTE: this is setting ``bitwise_flag == 4``
+
+        score_filter: bit 7 when set as 1, if the best
+            ``alignment_score >= score_filter``, (whatever bit 5 is set) the
+            function  will return the best alignment beginning position and cigar;
+            NOTE: this is setting ``bitwise_flag == 2``
+
+        best_idxs: bit 8 when set as 1, (whatever bit 5, 6 or 7 is setted) the
```

Comment:
Does `(whatever bit 5, 6 or 7 is setted)` just mean that `best_idxs` is applied if bit 8 is set irrespective of the value of bits 5-7?

Reply:
This is using language from the upstream documentation.  I have improved it a bit, but and will update this like again, but I think the language here could use another pass in a subsequent PR

Reply:
pushed